File: frontend/src/api/permission.ts
import request from '@/utils/request'

export interface Permission {
  permissionId: string
  permissionCode: string
  permissionName: string
  resource: string
  action: string
  description: string
  createdAt: string
  updatedAt: string
}

export interface CreatePermissionDto {
  permissionCode: string
  permissionName: string
  resource: string
  action: string
  description: string
}

// 获取权限列表
export function getPermissionsApi() {
  return request({
    url: '/api/permissions',
    method: 'get'
  })
}

// 获取权限详情
export function getPermissionDetailApi(id: string) {
  return request({
    url: `/api/permissions/${id}`,
    method: 'get'
  })
}

// 创建权限
export function createPermissionApi(data: CreatePermissionDto) {
  return request({
    url: '/api/permissions',
    method: 'post',
    data
  })
}

// 更新权限
export function updatePermissionApi(id: string, data: CreatePermissionDto) {
  return request({
    url: `/api/permissions/${id}`,
    method: 'put',
    data
  })
}

// 删除权限
export function deletePermissionApi(id: string) {
  return request({
    url: `/api/permissions/${id}`,
    method: 'delete'
  })
}

File: backend/src/main/java/com/testflowai/entity/Permission.java
package com.testflowai.entity;

import lombok.Data;
import java.io.Serializable;
import java.util.Date;

/**
 * 权限实体类
 * 对应数据库表：t_permission
 * @author TestFlowAI
 */
@Data
public class Permission implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 权限 ID (UUID)
     */
    private String permissionId;

    /**
     * 权限代码
     */
    private String permissionCode;

    /**
     * 权限名称
     */
    private String permissionName;

    /**
     * 资源类型
     */
    private String resource;

    /**
     * 操作类型
     */
    private String action;

    /**
     * 描述
     */
    private String description;

    /**
     * 创建时间
     */
    private Date createdAt;

    /**
     * 更新时间
     */
    private Date updatedAt;

    /**
     * 删除时间
     */
    private Date deletedAt;

    /**
     * 逻辑删除：0-未删除，1-已删除
     */
    private Integer deleted;

    /**
     * 创建者
     */
    private String createdBy;

    /**
     * 更新者
     */
    private String updatedBy;

    /**
     * 删除者
     */
    private String deletedBy;
}
